subsys/net/ip/ipv4.c

```diff
@@ -147,7 +151,16 @@ int net_ipv4_parse_hdr_options(struct net_pkt *pkt,
 				return -EINVAL;
 			}
 
+			if (opt_len < 2U) {
+				return -EINVAL;
+			}
+
 			opt_len -= 2U;
+
+			if (opts_len < 1U) {
+				return -EINVAL;
+			}
+
```

Comment:
Wouldn't it make sense to combine these so that the first test is `if (opt_len < 3U)`?

Reply:
And actually, why do you have two separate decrements? Perhaps a code comment to explain it?

Reply:
both are different opt_len, opt**s**_len

Reply:
Oh, right. Sorry - missed that